File: modules/agent/src/models/EmployeeCountRange.py
"""EmployeeCountRange model for organization size brackets."""

from sqlalchemy import Column, BigInteger, Text, Integer, DateTime
from sqlalchemy.sql import func
from models import Base


class EmployeeCountRange(Base):
    """Lookup table for employee count brackets (e.g., 1-10, 11-50)."""

    __tablename__ = "EmployeeCountRange"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    label = Column(Text, nullable=False, unique=True)
    min_value = Column(Integer, nullable=True)
    max_value = Column(Integer, nullable=True)
    display_order = Column(Integer, nullable=False, default=0)
    created_at = Column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

File: modules/agent/internal/models/EmployeeCountRange.go
package models

import "time"

type EmployeeCountRange struct {
	ID           int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Label        string    `gorm:"uniqueIndex;not null" json:"label"`
	MinValue     *int      `json:"min_value"`
	MaxValue     *int      `json:"max_value"`
	DisplayOrder int       `gorm:"not null;default:0" json:"display_order"`
	CreatedAt    time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt    time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (EmployeeCountRange) TableName() string {
	return "Employee_Count_Range"
}
